src/middlewared/middlewared/plugins/update_/download.py

```diff
@@ -38,9 +38,9 @@ def download_update(self, job, train, version, location, progress_proportion):
 
             job.set_progress(0, "Retrieving update manifest")
 
-            update_status = self.middleware.call_sync('update.status')
+            update_status = self.middleware.call_sync('update.status_internal', True)
             if update_status['error']:
-                raise CallError(f'Error retrieving update status: {update_status["error"]}')
+                raise CallError(f'Error retrieving update status: {update_status["error"]}', errno.ENOPKG)
```

Comment:
Can't status return an `error` that is possibly not related to ENOPKG? e.g. cant it return and error about REBOOT_REQUIRED? and others

Reply:
@william-gr what error code do you propose? You are correct, it might not be related to package versioning.

I don't believe we'll be able to classify all our custom error strings into pre-existing errno codes.

Reply:
The goal is not only avoid showing tracebacks but provide meaningful handling whenever possible.

e.g. how we handle ENETUNREACH might be different by directing the user to network configuration and/or give clues on what might be wrong. ECONNRESET or ETIMEDOUT we can infer possible intermittent network issues and tell him to try again later. Which is not really what is happening here as it stands.

Reply:
I added new error codes for this purpose. https://github.com/truenas/api_client/compare/NAS-137343?expand=1 is also necessary.